### ISSUE TITLE
How about to support Makefile variable pattern $(VARIABLE) in database-connection-string rule

### ISSUE THREAD

**Issue by pankona** [secretlint/secretlint#1190]
## Summary

How about to support Makfile variable patter `$(VAR)` in database-connection-string rule?
This must improve compatibility with Make-based projects.

### Basic example

My Makefile includes DB connection information like this:
```makefile
DB_URL = postgres://$(PGUSER):$(PGPASSWORD)@$(PGHOST)/$(PGDATABASE)
```

This Makefile is warned by secretlint. It seems such variable string like `$(PGPASSWORD)` is not included in isVariableLikeString function.
https://github.com/secretlint/secretlint/blob/ce6ac7da0c7593d47b712438b0e302d7b3631325/packages/%40secretlint/secretlint-rule-database-connection-string/src/index.ts#L62-L70

I'm glad if this is allowed as another variables pattern.
Here's example.
```ts
function isVariableLikeString(str: string): boolean {
    const variablePatterns = [
        /\$\{[^}]{1,50}\}/, // ${var}
        /\{\{[^}]{1,50}\}\}/, // {{var}}
        /\{[^}]{1,50}\}/, // {var}
        /%[A-Z_]{1,30}%/, // %VAR%
        /\$[A-Z_]{1,30}/, // $VAR
        /\$\([^)]{1,50}\)/, // $(var) - *** Add line like this? ***
    ];

    return variablePatterns.some((pattern) => pattern.test(str));
}
```

### Motivation

- I think it is common to have database connection strings in a Makefile.
- Supporting common variable patterns would make secretlint even more useful.
- Of course, it helps my project, I would be happy, too.

**Comment by azu** [comment]
I agree to allow it.

**Comment by azu** [comment]
https://github.com/secretlint/secretlint/releases/tag/v11.2.0
released 🎉

**Comment by pankona** [comment]
@azu Thank you so much! 🎉
